datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapWriter.java

```diff
@@ -102,24 +96,25 @@
     this.indexedCarbonColumns = indexedCarbonColumns;
   }
 
-  private String getIndexPath(long taskId) {
+  private String getIndexPath(String taskName) {
     if (isFineGrain) {
-      return genDataMapStorePathOnTaskId(identifier.getTablePath(), segmentId, dataMapName, taskId);
+      return genDataMapStorePathOnTaskId(identifier.getTablePath(), segmentId, dataMapName,
+          taskName);
     } else {
       // TODO: where write data in coarse grain data map
-      return genDataMapStorePathOnTaskId(identifier.getTablePath(), segmentId, dataMapName, taskId);
+      return genDataMapStorePathOnTaskId(identifier.getTablePath(), segmentId, dataMapName,
+          taskName);
     }
   }
 
   /**
    * Start of new block notification.
    */
-  public void onBlockStart(String blockId, long taskId) throws IOException {
+  public void onBlockStart(String blockId, String indexShardName) throws IOException {
```

Comment:
In this method, we should only initialize indexWriter once.
It means all blocks of this task will share the index writer.

Reply:
if indexWriter != null then not required to set it again.